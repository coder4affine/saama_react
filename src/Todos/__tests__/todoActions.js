import React from 'react';

import { render, fireEvent } from '@testing-library/react';
import TodoActions from '../todoActions';

describe('TodoActions', () => {
  const mockChangeStatus = jest.fn();

  const props = {
    changeStatus: mockChangeStatus,
  };

  afterEach(() => {
    // mockChangeStatus.mockClear();
    jest.clearAllMocks();
  });

  test('should render TodoActions', () => {
    const { container } = render(<TodoActions {...props} />);

    expect(container).toMatchSnapshot();
  });

  test('should check length of buttons', () => {
    const { getAllByRole } = render(<TodoActions {...props} />);

    expect(getAllByRole('button').length).toBe(3);
  });

  test('should click all button', () => {
    const { getByTestId } = render(<TodoActions {...props} />);

    fireEvent.click(getByTestId('btn-all'));

    expect(mockChangeStatus).toHaveBeenCalledTimes(1);
    expect(mockChangeStatus).toHaveBeenCalledWith('all');
  });

  test('should click pending button', () => {
    const { getByTestId } = render(<TodoActions {...props} />);

    fireEvent.click(getByTestId('btn-pending'));

    expect(mockChangeStatus).toHaveBeenCalledTimes(1);
    expect(mockChangeStatus).toHaveBeenCalledWith('pending');
  });

  test('should click complete button', () => {
    const { getByTestId } = render(<TodoActions {...props} />);

    fireEvent.click(getByTestId('btn-completed'));

    expect(mockChangeStatus).toHaveBeenCalledTimes(1);
    expect(mockChangeStatus).toHaveBeenCalledWith('completed');
  });
});
