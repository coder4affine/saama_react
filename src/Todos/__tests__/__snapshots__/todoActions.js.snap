// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`TodoActions should render TodoActions 1`] = `
<div>
  <div
    style="display: flex;"
  >
    <button
      data-testid="btn-all"
      style="flex: 1; justify-content: center; align-items: center;"
      type="button"
    >
      All
    </button>
    <button
      data-testid="btn-pending"
      style="flex: 1; justify-content: center; align-items: center;"
      type="button"
    >
      Pending
    </button>
    <button
      data-testid="btn-completed"
      style="flex: 1; justify-content: center; align-items: center;"
      type="button"
    >
      Completed
    </button>
  </div>
</div>
`;
